lib/schemapack.js

```diff
@@ -382,15 +381,6 @@ class SchemaPack {
       matchedEndpoints = [],
       jsonSchemaDialect = schema.jsonSchemaDialect;
 
-    // Only change the stack limit if the optimizeConversion option is true
```

Comment:
How much depth to use is not required in validateTransaction?

Reply:
@umeshp7 Fixed the failing tests. This other PR fixed them but fixed issue separately before other PR fixes it.

Reply:
@umeshp7 As during validation we can not set `optimizeConversion` option to `false`, the default behavior is making changes to `stackLimit`.

I have mentioned more details in RCA around this decision.

Reply:
Okay, please share that. I am curious because: If during conversion we decide the depth should be 6. And while validateTransaction, if that same depth is not considered and the option is false. It will fallback to 10 which might cause problems right?
@VShingala